src/views/docs-view/server.ts

```diff
@@ -168,6 +168,22 @@ export function getStaticGenerationFunctions<
 				pathParts.join('/')
 			)}`
 			const headings: AnchorLinksPluginHeading[] = [] // populated by anchorLinks plugin below
+			// Hide the version selector on TFE release notes page, e.g. /terraform/enterprise/<year>/<version> or /vault/docs/release-notes/1.13.0
+			const isReleaseNotesPage = (path) => {
+				return /(\/releases\/[0-9]{4}\/(v[0-9]{6}-\d+))/i.test(path) ||
+					/\/release-notes\/\d+[.]\d+[.](\d+|x)/i.test(path)
+					? true
+					: false
+			}
+			if (isReleaseNotesPage(currentPathUnderProduct)) {
+				options.hideVersionSelector = true
+			} else if (
+				!isReleaseNotesPage(currentPathUnderProduct) &&
+				options.hideVersionSelector === true
+			) {
+				// toggle version dropdown to visible when not on release notes page
+				options.hideVersionSelector = false
+			}
```

Comment:
Would you mind explaining this a little more? If `options.hideVersionSelector` is explicitly `true`, why are we overriding it?

Reply:
Happy to add more context here. This is to correct the situation I ran into where the version dropdown wouldn't populate after navigating to a non-release notes docs page.

For example, a user navigates to `/terraform/enterprise/releases/2024/202401-1` from the releases overview page and then clicks on a sidebar link, such as `Migrating to Terraform Cloud or Terraform Enterprise` and the version dropdown wouldn't appear on that page even though `/terraform/enterprise/migrate` is not a release notes page. This was happening because `options.hideVersionSelector = true` and wasn't getting reset to a falsy value on route change. 

Let me know if the logic here is still unclear.